Sources/SwiftSDKGenerator/Generator/SwiftSDKGenerator+Copy.swift

```
@@ -73,6 +73,11 @@ extension SwiftSDKGenerator {
         // But not in all containers, so don't fail if it does not exist.
         if case .ubuntu = targetDistribution {
           subpaths += [("\(targetTriple.archName)-linux-gnu", false)]
+
+          // Custom subpath for armv7
+          if targetTriple.archName == "armv7" {
+            subpaths += [("arm-linux-gnueabihf", false)]
```

Does Alpine or other Musl-base distributions use this path? I'd assume those support armv7, just thinking how we can future-proof these codepaths to make it easier to support Musl in the future.

---

I did a little looking at some Alpine containers and seems like it doesn't follow the Debian scheme of multi-arch by putting stuff at /usr/lib/<arch files>. It looks that it just has the basic ld and libc stuff in /lib:

```bash
# x86_64
/ # ls /lib/*.so*
/lib/ld-musl-x86_64.so.1    /lib/libc.musl-x86_64.so.1

# armhf
~ # ls /lib/*.so*
/lib/ld-musl-armhf.so.1 /lib/libc.musl-armhf.so.1
```

So I guess what would really have to change is the interpreter path to point to something lib `/lib/ld-musl-<arch>.so.1`:

```
extension Triple {
  var interpreterPath: String {
    switch self.archName {
      case "x86_64": return "/lib64/ld-linux-x86-64.so.2"
      case "aarch64": return "/lib/ld-linux-aarch64.so.1"
      case "armv7": return "/lib/ld-linux-armhf.so.3"
      default: fatalError("unsupported architecture \(self.archName)")
    }
  }
}
```

The `arm-linux-gnueabihf` subpath is only for Debian it seems!

---

Can we make it conditional on Debian somehow and issue a warning or whatever else is suitable for other unrecognized armv7 distributions?

---

Do you mean to throw an error if someone tries to create an armv7 Swift SDK when specifying .rhel as the distribution? The block of code you mention above is already under a .ubuntu conditional to only add these special directories for ubuntu distributions.

---

Right now if I do something like this:

```bash
$ swift run swift-sdk-generator make-linux-sdk --swift-version 6.0.3-RELEASE --host x86_64-unknown-linux-gnu --target armv7-unknown-linux-gnueabihf --linu
x-distribution-name rhel --with-docker
```

I just get a Docker error, which of course means there is no container available for rhel-ubi9 for armv7:

```bash
Launching a Docker container to copy Swift SDK for the target triple from it...
Error: process exited non-zero: exit(125)
```

So perhaps this situation can be detected up front, and we can throw an error saying that RHEL UBI9 does not support armv7.

---

WDYT of logging a warning message at least when non-Ubuntu codepath is triggered for armv7 to point users in the right direction?